common/app/layout/Sublink.scala

```diff
@@ -37,22 +37,31 @@ case class EditionalisedLink(
 }
 
 object Sublink {
-  def fromFaciaContent(faciaContent: PressedContent): Sublink =
+  def fromFaciaContent(faciaContent: PressedContent): Sublink = {
+    val storyContent: Option[PressedStory] = faciaContent.properties.maybeContent
+    val pillar: Pillar = storyContent.flatMap(_.metadata.pillar).getOrElse(Pillar.News)
+    val contentType: DotcomContentType = storyContent.flatMap(_.metadata.`type`).getOrElse(DotcomContentType.Article)
```

Comment:
This logic is already there https://github.com/guardian/frontend/blob/6cc9d378a14468ff325dc84fcf7e4469eaf2c3c7/common/app/layout/Sublink.scala#L349-L351

Would it make sense to refactor it into `Pillar` and `DotcomContentType` object in order to avoid duplication?

Reply:
why is it going to be ripped out?

Reply:
sorry @TBonnin, you're right, ignore me

Reply:
@TBonnin I'm not sure how I can reuse that logic from the case class `ContentCard` in this method`fromFaciaContent`? This method accepts the argument `faciaContent` which is a type `PressedContent`.

The`PressedContent` type has no reference to a `ContentCard` in it's properties. What it does have is `properties.maybeContent` which we can derive an `Option[PressedStory]` from, and eventually a `Pillar` and `ContentType` from this, in the same way we do in the case class `ContentCard`.